packages/expo-cli/src/commands/upgrade.ts

```diff
@@ -465,21 +466,32 @@ export async function upgradeAsync(
       await maybeUpgradeEmulatorAsync();
     }
   } else if (!targetSdkVersion) {
-    if (program.nonInteractive) {
-      log.warn(
-        `You provided the target SDK version value of ${targetSdkVersionString} which does not seem to exist. Upgrading anyways because the command is running in nonInteractive mode.`
-      );
-    } else {
-      // If they provide an apparently unsupported sdk version then let people try
-      // anyways, maybe we want to use this for testing alpha versions or
-      // something...
-      const answer = await confirmAsync({
-        message: `You provided the target SDK version value of ${targetSdkVersionString} which does not seem to exist. But hey, I'm just a program, what do I know. Do you want to try to upgrade to it anyways?`,
-      });
+    // This is useful when testing the beta internally, before actually
+    // releasing it as a public beta. At this point, we won't have "beta" set on
+    // the versions endpoint and so Versions.releasedSdkVersionsAsync will not
+    // return the beta version, even with the EXPO_BETA flag set.
+    if (getenv.boolish('EXPO_BETA', false)) {
+      const allSdkVersions = await Versions.sdkVersionsAsync();
+      targetSdkVersion = allSdkVersions[targetSdkVersionString];
+    }
 
-      if (!answer) {
-        return;
-      }
+    // If we still don't have a version, even after searching through unreleased versions
```

Comment:
this changes existing behavior on `expo upgrade` for unknown sdk versions. these would not actually work anyways, so just actually bail out instead